soccer/gameplay/tactics/defense.py

```diff
@@ -78,435 +70,297 @@ def debug(self, value):
             if self.debug:
-                for handler in threat.assigned_handlers:
-                    # handler.robot.add_text("Marking: " + str(threat.source), constants.Colors.White, "Defense")
-                    main.system_state().draw_circle(handler.move_target, 0.02,
-                                                    constants.Colors.Blue,
-                                                    "Defense")
-
-                # draw some debug stuff
-                if threat.best_shot_window is not None:
-                    # draw shot triangle
-                    pts = [threat.pos,
-                           threat.best_shot_window.segment.get_pt(0),
-                           threat.best_shot_window.segment.get_pt(1)]
-                    shot_color = (255, 0, 0, 150)  # translucent red
-                    main.system_state().draw_polygon(pts, shot_color,
-                                                     "Defense")
-                    main.system_state().draw_segment(
-                        threat.best_shot_window.segment, constants.Colors.Red,
-                        "Defense")
-
-                    self.win_eval.excluded_robots.clear()
-                    _, best_window = self.win_eval.eval_pt_to_our_goal(
-                        threat.pos)
-                    if best_window is not None:
-                        chance = best_window.shot_success
-                    else:
-                        chance = 0.0
-
-                    main.system_state().draw_text(
-                        "Shot: " + str(int(threat.shot_chance * 100.0)) +
-                        "% / " + str(int(chance * 100)) + "%",
-                        shot_line.center(), constants.Colors.White, "Defense")
-
-                # draw pass lines
-                if idx > 0:
-                    pass_line = robocup.Segment(main.ball().pos, threat.pos)
+                main.system_state().draw_line(shot_line, constants.Colors.Red,
+                                              "Defense-Shot Line")
+                main.system_state().draw_text(
+                    "Shot: " + str(int(shot_chance * 100.0)), threat[0],
+                    constants.Colors.White, "Defense-Shot Percent")
+
+                # Other threats besides ball
+                if threat_idx > 0:
+                    pass_line = robocup.Segment(main.ball().pos, threat[0])
                     main.system_state().draw_line(
-                        pass_line, constants.Colors.Red, "Defense")
-                    main.system_state().draw_text(
-                        "Pass: " + str(int(threat.ball_acquire_chance * 100.0))
-                        + "%", pass_line.center(), constants.Colors.White,
-                        "Defense")
+                        pass_line, constants.Colors.Red, "Defense-Pass Line")
+
+            #keep defenders from occupying the same spot
+            #only matters if there are 2 defenders (and the goalie)
+            if len(handlers) == 3:
```

Comment:
Is there any reason you can't go over every combination of 2 points here instead of just using the two defenders? Does that break the goalie? I think thats a bit cleaner and less likely to break.

Reply:
I could but since this is the only case that matters it seems like a waste of time ¯\_(ツ)_/¯